File: app/src/main/java/padev/badhabits/Data/IDataAccessObject.java
package padev.badhabits.Data;

import java.util.ArrayList;

public interface IDataAccessObject {

    void insertData(AbstractData data);

    AbstractData selectData(int id);

    ArrayList<AbstractData> selectAllData();

    int updateData(AbstractData data);

    void deleteData(int id);

    void deleteAllData();
}

File: app/src/main/java/padev/badhabits/Data/IDataAccessObject.kt
package padev.badhabits.Data;

import java.util.ArrayList;

public interface IDataAccessObject {

    fun insertData(data: AbstractData)

    fun selectData(id: Int): AbstractData

    fun selectAllData(): ArrayList<AbstractData>

    fun updateData(data: AbstractData): Int

    fun deleteData(id: Int)

    fun deleteAllData()
}